toolkit/tools/imagegen/configuration/networkconfig.go

```diff
@@ -0,0 +1,368 @@
+	if err = n.validateIPAddress(netmask); err != nil {
+		return fmt.Errorf("Invalid input for --netmask: %s", err)
+	}
+
+	gateway := strings.Trim(n.GateWay, " ")
+	if err = n.validateIPAddress(gateway); err != nil {
+		return fmt.Errorf("Invalid input for --gateway: %s", err)
+	}
+
+	for _, nameserver := range n.NameServer {
+		if err = n.validateIPAddress(strings.Trim(nameserver, " ")); err != nil {
+			return fmt.Errorf("Invalid input for --nameserver: %s", err)
+		}
+	}
+
+	return
+}
+
+// DeviceIsValid returns an error if the Device name is empty
+func (n *Network) DeviceIsValid() (err error) {
+	if strings.TrimSpace(n.Device) == "" {
+		return fmt.Errorf("Invalid input for --device (), device cannot be empty")
+	}
+	return
+}
+
+func findBootIfValue() (deviceAddr string) {
+	const (
+		kernelArgFile = "/proc/cmdline"
+		startIndex    = 10
+	)
+
+	content, err := os.ReadFile(kernelArgFile)
+	if err != nil {
+		logger.Log.Errorf("failed to read from /proc/cmdline")
+		return
+	}
+
+	// Find the location of BOOTIF
+	kernelArgs := strings.Split(string(content), " ")
+	for _, kernelArg := range kernelArgs {
+		if strings.Contains(kernelArg, "BOOTIF") {
+			rawDeviceAddr := []byte(kernelArg[startIndex:len(kernelArg)])
+
+			// Change the cmdline MAC address into valid format
+			for i, _ := range rawDeviceAddr {
+				if rawDeviceAddr[i] == '-' {
+					rawDeviceAddr[i] = ':'
+				}
+			}
+			deviceAddr = string(rawDeviceAddr)
+			return
+		}
+	}
+
+	return
+}
+
+func checkNetworkDeviceAvailability(networkData Network) (deviceName string, err error) {
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return
+	}
+
+	if networkData.Device == "bootif" {
+		networkData.Device = findBootIfValue()
+	}
+
+	for _, iface := range ifaces {
+		if networkData.Device == iface.Name {
+			deviceName = networkData.Device
+			return
+		} else {
+			ifaceAddr := strings.TrimSpace(iface.HardwareAddr.String())
+
+			if strings.Contains(networkData.Device, ":") {
+				networkData.Device = strings.TrimSpace(networkData.Device)
+
+				if ifaceAddr != "" && strings.EqualFold(ifaceAddr, networkData.Device) {
+					deviceName = iface.Name
+					return
+				}
+			}
+		}
+	}
+
+	return
+}
+
+func populateMatchSection(networkData Network, fileName, deviceName string) (err error) {
+	const (
+		id        = "[Match]\n"
+		NameField = "Name="
+	)
+
+	matchSection := id + NameField + deviceName + "\n"
+	err = file.Append(matchSection+"\n", fileName)
+	if err != nil {
+		logger.Log.Infof("Failed to write [Match] section: %s", err)
+	}
+
+	return
+}
+
+func populateNetworkSection(networkData Network, fileName string) (err error) {
+	const (
+		id           = "[Network]\n"
+		ipField      = "Address="
+		gateWayField = "Gateway="
+		dnsField     = "DNS="
+	)
+
+	networkSection := id
+
+	// Currently only supports static IP setting
+	// Update IP and netmask
+	netMask := strings.Trim(networkData.NetMask, " ")
+	ip := strings.Trim(networkData.NetMask, " ")
```

Comment:
Why `strings.Trim` instead of `strings.TrimSpace`? Do we want to keep other whitespaces? If so, it might be worth mentioning that in a comment.